libs/shared/console-shared/src/lib/cluster-settings/ui/cluster-general-settings/cluster-general-settings.spec.tsx

```diff
@@ -0,0 +1,33 @@
+import { render } from '__tests__/utils/setup-jest'
+import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
+import ClusterGeneralSettings, { ClusterGeneralSettingsProps } from './cluster-general-settings'
+
+describe('ClusterGeneralSettings', () => {
+  const props: ClusterGeneralSettingsProps = {
+    fromDetail: false,
+  }
+  it('should render successfully', () => {
+    const { baseElement } = render(wrapWithReactHookForm(<ClusterGeneralSettings {...props} />))
+    expect(baseElement).toBeTruthy()
+  })
+
+  it('should submit the form on click', async () => {
+    const { getByTestId } = render(
+      wrapWithReactHookForm(<ClusterGeneralSettings {...props} />, {
```

Comment:
Since few weeks, you can pass the type that defaultValues will be by doing : `wrapWithReactHookForm< ClusterGeneralData>` this is convenient for autocompletion and type checks